src/EReg/ElectronicRegulator.cpp

```diff
@@ -57,55 +41,61 @@ void flow(Comms::Packet packet) {
     startFlow = packet.data[0];
 }
 
-void fullOpen(Comms::Packet packet) {
+void stopFlow(Comms::Packet packet) {
 
 }
 
 void setPIDConstants(Comms::Packet packet) {
 
 }
 
+void abort(Comms::Packet packet) {
 
-// valve angle based on pressure setpoint
-PIDController outerController(1.50, 2.25e-6, 0.125, MIN_ANGLE, MAX_ANGLE);
-// motor angle based on valve setpoint
-
-// pressure against motor during flow different than tuning seperately? 
-PIDController innerController(11.5, 1.5e-6, 0.35e-6, MIN_SPD, MAX_SPD);
+}
 
 void setup() {
-    Serial.begin(9600);
 
     Comms::registerCallback(0, zero);
     Comms::registerCallback(1, setPressureSetpoint);
     Comms::registerCallback(2, flow);
-    Comms::registerCallback(3, fullOpen);
+    Comms::registerCallback(3, stopFlow);
+    Comms::registerCallback(3, setPIDConstants);
+    Comms::registerCallback(3, abort);
 }
 
 void loop() {
     Comms::processWaitingPackets();
 
     float angle = encoder.read();
-    float motorAngle = Util::encoderToAngle(angle);
+    float motorAngle = angle;
     float potAngle = Util::readPot();
     float HPpsi = Util::voltageToHighPressure(analogRead(HAL::hpPT));
-    float LPpsi = Util::voltageToPressure(analogRead(HAL::lpPT));
-    float InjectorPT = Util::voltageToPressure(analogRead(HAL::injectorPT));
+    float LPpsi = Util::voltageToLowPressure(analogRead(HAL::lpPT));
+    float InjectorPT = Util::voltageToLowPressure(analogRead(HAL::injectorPT));
 
     if (startFlow) {
         //Compute Inner PID Servo loop
-        float speed = innerController.update(motorAngle - angle_setpoint);
+        float speed = innerController->update(motorAngle - angle_setpoint);
 
         //Compute Outer Pressure Control Loop
-        float angle_setpoint = outerController.update(LPpsi - pressure_setpoint);
+        angle_setpoint = outerController->update(LPpsi - pressure_setpoint);
         angle_setpoint += Util::compute_feedforward(pressure_setpoint, HPpsi);
 
-        runMotors(speed);
+        Util::runMotors(speed);
     }
 
     //send data to AC
     if (micros() - lastPrint > 1000) {
         Comms::Packet packet = {.id = 85};
+        //TODO split into two temelemtry packets
+        Comms::packetAddFloat(&packet, Util::p_outer);
+        Comms::packetAddFloat(&packet, Util::i_outer);
+        Comms::packetAddFloat(&packet, Util::d_outer);
+        Comms::packetAddFloat(&packet, Util::p_inner);
+        Comms::packetAddFloat(&packet, Util::i_inner);
+        Comms::packetAddFloat(&packet, Util::d_inner);
+        Comms::packetAddFloat(&packet, pressure_setpoint);
+        Comms::packetAddFloat(&packet, angle_setpoint);
         Comms::packetAddFloat(&packet, HPpsi);
```

Comment:
add:
speed
angle
outer k_p, k_i, k_d

remove:
inner p, i, d

Reply:
by outer k_p, k_i, k_d do you mean the constants or the terms in the output equation?

Reply:
k_p etc. are the constants. So we'll send both the constants and the terms